tests/conftest.py

```diff
@@ -13,29 +13,27 @@ def tmp_dir(tmp_path_factory):
     tmp_path = tmp_path_factory.getbasetemp() / "image_cache"
     tmp_path.mkdir()
 
-    # download the (only) test images via git's sparse-checkout
-    current_path = os.getcwd()
-    os.chdir(tmp_path)
-    git_version = subprocess.run(
-        ["git", "--version"], stdout=subprocess.PIPE
-    ).stdout.decode("utf-8")[12:18]
-    major, minor, _ = git_version.split(".")
-
-    if int(major) == 2 and int(minor) < 30:
-        # --sparse was introduced in git 2.30.0 (I think)
-        os.system(
-            "git clone --depth 1 --filter=blob:none --no-checkout https://github.com/imageio/imageio-binaries.git ."
-        )
-        os.system("git sparse-checkout init --cone")
-        os.system("git sparse-checkout set test-images")
-    else:
-        # sparse checkout on GH Actions
-        os.system(
-            "git clone --sparse --filter=blob:none https://github.com/imageio/imageio-binaries.git ."
-        )
-        os.system("git sparse-checkout init --cone")
-        os.system("git sparse-checkout set test-images")
-    os.chdir(current_path)
+    if not os.path.isfile(tmp_path / "chelsea.png"):
```

Comment:
I don't think this branch is necessary. This fixture is run once at the beginning of a test session (session scoped) and tmp_path will be a clean  directory.

Reply:
I did see it run more often once, but maybe that was because I put a `1/0` in it. Does it make sense that it will then try it again and again?

Reply:
I think so. I never dug into it explicitly, but as far as I understand, a cache entry for a fixture is only cached if it executes successfully. If it fails (e.g. because of a div/0 exception) then it should be run repeatedly since there is no cache entry to reuse.